src/lib/OpenEXRCore/internal_ht_common.cpp

```diff
@@ -6,34 +6,138 @@
 #include "internal_ht_common.h"
 #include <vector>
 #include <string>
+#include <string.h>
 #include <cassert>
+#include <algorithm>
+#include <cctype>
+
+const std::string RED_CH_FULLNAME = "red";
+const std::string GREEN_CH_FULLNAME = "green";
+const std::string BLUE_CH_FULLNAME = "blue";
+
+struct RGBChannelParams
+{
+    const char* r_suffix;
+    const char* g_suffix;
+    const char* b_suffix;
+    int         r_index;
+    int         g_index;
+    int         b_index;
+    const char* prefix;
+    size_t      prefix_len;
+};
+
+static inline bool areEqual(const char* a, const char* b) {
+#ifdef _MSC_VER
+    return _stricmp (a, b) == 0;
+#else
+    return strcasecmp (a, b) == 0;
+#endif
+}
 
 bool
 make_channel_map (
-    int channel_count, exr_coding_channel_info_t* channels, std::vector<CodestreamChannelInfo>& cs_to_file_ch)
+    int                                 channel_count,
+    exr_coding_channel_info_t*          channels,
+    std::vector<CodestreamChannelInfo>& cs_to_file_ch)
 {
-    int r_index = -1;
-    int g_index = -1;
-    int b_index = -1;
+    /** Heuristic detection of RGB channels so that the JPEG 2000 Reversible
+      * Color Transform (RCT), a decorrelation transform, can be applied.
+      *
+      * RGB channels are present if the either (a) the names of the channels in
+      * their entirety or (b) following the "." character match one of the
+      * triplets defined in {params}. Order of the channels and case of the
+      * channel names are ignored.
+      *
+      * Example 1: "main.b", "main.g", "main.r" match
+      * Example 2: "MainR", "MainG", "MainB" do not match
+      * Example 3: "R", "B", "B" match
+      * Example 4: "red", "green", "blue" match
+      */
 
-    cs_to_file_ch.resize(channel_count);
+    RGBChannelParams params[] = {
+        {"r", "g", "b", -1, -1, -1, NULL, 0},
+        {"red", "green", "blue", -1, -1, -1, NULL, 0},
+        {"red", "grn", "blu", -1, -1, -1, NULL, 0}};
```

Comment:
I still wonder if there's a tool that produces red, gin, blu, in other words, what's the motivation or including them?

Reply:
My opinion is that it suggests to the reader of the code that those abbreviations are expected as legal, but we don't do that anywhere else so I prefer to elide them without something to ground them.